content/250-postgres/300-database/750-serverless-driver.mdx

```diff
@@ -0,0 +1,99 @@
+---
+title: "Serverless driver"
+metaTitle: "Serverless driver"
+metaDescription: "The Prisma serverless driver enables direct access to Prisma Postgres via HTTP."
+tocDepth: 3
+toc: true
+sidebar_class_name: early-access-badge
+---
+
+The serverless driver for Prisma Postgres is a lightweight and minimal client library that can talk to Prisma Postgres using raw SQL. You can use it via the [`@prisma/ppg`](https://github.com/prisma/ppg-client/tree/main) npm package.
+
+## Overview
+
+### Installation
+
+Install the serverless driver via npm:
+
+```terminal
+npm install @prisma/ppg
+```
+
+### Usage
+
+The recommended API for most users is the `ppg` function, which returns a high-level SQL client implemented as a template literal tag function:
+
+```ts
+import { ppg } from "@prisma/ppg";
+
+const sql = ppg("prisma+postgres://accelerate.prisma-data.net/?api_key=...");
+
+const userId = 1;
+const posts = await sql`SELECT * FROM "Post" WHERE "userId" = ${userId}`;
+```
+
+## API reference
+
+### `ppg`
+
+The `ppg` function returns a high-level SQL client implemented as a template literal tag function:
+
+```ts
+function ppg(connectionString: string, deserialize?: Deserialize): Sql;
+```
+
+The returned `Sql` function:
+
+- Accepts a SQL statement as a template literal. If this contains any interpolated values, these are automatically converted to SQL parameters in order to prevent SQL injection attacks (see below for an example).
+- Returns the data as an array of objects mirroring the structure of your Prisma models.
+
+#### Arguments
+
+The `ppg` function accepts the following arguments:
+
+| Name               | Type          | Required | Description                                                                                                                                                                                   |
+| :----------------- | :------------ | :------- | :-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `connectionString` | `string`      | **Yes**  | The connection string for your Prisma Postgres instance.                                                                                                                                      |
+| `deserialize`      | `Deserialize` | No       | Custom deserializer function which takes a column type OID and a raw value and returns a mapped value. It's type is defined as `type Deserialize = (value: unknown, oid: unknown) => unknown` |
+
+#### Usage
+
+```ts
+import { ppg } from "@prisma/ppg";
+import type { User, Post } from "@prisma/client";
```

Comment:
Let's avoid using generated types in this docs page, as it gives the impression that you need to use the CLI to generate code.

Reply:
![CleanShot 2025-05-16 at 16 40 56@2x](https://github.com/user-attachments/assets/36a8e1c0-45e4-4bef-828e-40b5c0dfc0f2)

This is an example of what V0 actually does when using Neon

Reply:
ah, that's a good point! my goal was to show the compatibility of the Prisma types with the returned values but I see how  that may give a wrong impression. I still think there's value in showing the compabiltiy but don't want to overemphasize it, how about all default examples don't use that and then we do a dedicated one talking about "compatibility with generated Prisma types" (or smth like that) to demonstrate that. 

or do you think this is too misleading and even with that context there wouldn't be much value in showing that?

Reply:
It's not currently compatible with Prisma types though (except in trivial cases)

Reply:
e.g. dates and decimals would definitely be incompatible

Reply:
ah, then I'll remove that entirely